From 9744184f88af9230a15bf213262e78e94d0db441 Mon Sep 17 00:00:00 2001
From: "haolong.zhang" <haolong.zhang@ck-telecom.com>
Date: Wed, 10 Dec 2014 09:34:56 +0800
Subject: [PATCH] [Feature] port rtxxx wifi dongle step 4 : AP mode config
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

[id]::72512
[Type]
   FEATURE
[Products]
	EYES_BASELINE
[Requirement]
   N/A

[Solution Description]
   N/A

[Other Info]
   N/A

 要提交的变更：
	修改:         frameworks/base/core/res/res/values/config.xml
	修改:         hardware/libhardware_legacy/wifi/wifi.c
	修改:         mediatek/config/mt6595/init.rc
	修改:         system/netd/SoftapController.h

Change-Id: I4d363cfb82584e14a670673dbd7f1723fad31a41
---
 frameworks/base/core/res/res/values/config.xml | 2 +-
 hardware/libhardware_legacy/wifi/wifi.c        | 7 ++++++-
 mediatek/config/mt6595/init.rc                 | 9 +++++++++
 system/netd/SoftapController.h                 | 2 +-
 4 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/frameworks/base/core/res/res/values/config.xml b/frameworks/base/core/res/res/values/config.xml
index 302be15..8001690 100755
--- a/frameworks/base/core/res/res/values/config.xml
+++ b/frameworks/base/core/res/res/values/config.xml
@@ -255,7 +255,7 @@
          Wifi interfaces.  If the device doesn't want to support tethering over Wifi this
          should be empty.  An example would be "softap.*" -->
     <string-array translatable="false" name="config_tether_wifi_regexs">
-        <item>"ap\\d"</item>
+        <item>"wlan\\d"</item>
     </string-array>
 
     <!-- List of regexpressions describing the interface (if any) that represent tetherable
diff --git a/hardware/libhardware_legacy/wifi/wifi.c b/hardware/libhardware_legacy/wifi/wifi.c
index 9710d33..d4f0e93 100644
--- a/hardware/libhardware_legacy/wifi/wifi.c
+++ b/hardware/libhardware_legacy/wifi/wifi.c
@@ -689,7 +689,12 @@ int wifi_connect_on_socket_path(const char *path)
     char supp_status[PROPERTY_VALUE_MAX] = {'\0'};
 
     ALOGD("enter -->%s path=%s\n", __func__, path);
-
+	if (strcmp(path, "/data/misc/wifi/hostapd/ap0") ==0)
+	{
+		ALOGD("soft ap com socket come,old path = %s, change to rtlink wlan0 socket\n", path);
+		path = "/data/misc/wifi/hostapd/wlan0";
+		ALOGD("now sockect path = %s\n", path);
+	}
 #ifdef CONFIG_HOTSPOT_MGR_SUPPORT
     if (property_get(FW_PATH_PROP_NAME, current_fw_path, NULL)
             && strcmp(current_fw_path, WIFI_DRIVER_FW_PATH_AP) == 0) {
diff --git a/mediatek/config/mt6595/init.rc b/mediatek/config/mt6595/init.rc
index e48c010..8491bac 100755
--- a/mediatek/config/mt6595/init.rc
+++ b/mediatek/config/mt6595/init.rc
@@ -1408,6 +1408,15 @@ service wpa_supplicant /system/bin/wpa_supplicant \
 	disabled
 	oneshot
 
+service ap_daemon /system/bin/wpa_supplicant \
+	-iwlan0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -e/data/misc/wifi/entropy.bin -puse_p2p_group_interface=1 -ddd
+	#   we will start as root and wpa_supplicant will switch to user wifi
+	#   after setting up the capabilities required for WEXT
+	#   user wifi
+	#   group wifi inet keystore
+	class main
+	disabled
+	oneshot
 
 #service wlanLoader /system/bin/wlan_loader
 #    user root
diff --git a/system/netd/SoftapController.h b/system/netd/SoftapController.h
index 7063067..65cebbd 100644
--- a/system/netd/SoftapController.h
+++ b/system/netd/SoftapController.h
@@ -25,7 +25,7 @@
 #define AP_BSS_STOP_DELAY	500000
 #define AP_SET_CFG_DELAY	500000
 #define AP_DRIVER_START_DELAY	800000
-#define AP_CHANNEL_DEFAULT	6
+#define AP_CHANNEL_DEFAULT	9
 
 class SoftapController {
 public:
-- 
1.9.1

